daemon/listeners/listeners_linux.go

```diff
@@ -32,27 +36,34 @@ func Init(proto, addr, socketGroup string, tlsConfig *tls.Config) ([]net.Listene
 		}
 		ls = append(ls, l)
 	case "unix":
-		gid, err := lookupGID(socketGroup)
+		gid := os.Getgid()
+		l, err := sockets.NewUnixSocket(addr, gid)
```

Comment:
nit: could simplify to

```suggestion
		l, err := sockets.NewUnixSocket(addr, os.Getgid())
```

However, with the current code, it looks like we'll be creating the socket to have the `root` group, not the `docker` group, correct? (I expect the process to be running as `root:root`, and not `docker` to be the primary group), and then do a `chown` after, so we'll always be `chown`-ing twice

Note that more recent versions of the go-connections package have a `NewUnixSocketWithOpts`, which could possibly avoid that; https://github.com/docker/go-connections/blob/09f47925228a97335b2462178c79669712efe44e/sockets/unix_socket.go#L81

With that one, we could pass a custom functional argument that does the `chown`

Reply:
I think the file UID doesn't matter much. What I mean is, whether it's `root` or `docker`, it's fine either way.

What matters is GID.